### PR TITLE
Implement asynchronous wait for broker startup.

### PR DESCRIPTION
**Pull request by soenke-tippelmann**

### CODE REVIEW

File: lib/druid.js [hunk 1]
```diff
@@ -65,10 +67,11 @@ function asyncMap(array, iterator, done) {
 function getDataSources(nodeInfo, callback) {
   var url = nodeInfo.url + '/druid/v2/datasources'
 
-  request({
-    url:  url,
-    json: true
-  }, onresponse)
+  async.retry(
+      {times: 120, interval: 1000},
+      _.partial(request, {url: url, json: true}),
```
Comment (dohse) [inline lib/druid.js]: Consider avoiding partial here and use a normal closure.

Comment (soenke-tippelmann) [reply lib/druid.js]: Thanks for the feedback. I'm not exactly sure what you mean by "using a normal closure". Should I write a function like
```
function(cb){request({url: url, json: true}, cb);}
```
instead? What would be the advantage?

Comment (dohse) [reply lib/druid.js]: Yeah, I mean that. I believe inline-functions are easier to read and you do not have to include lodash. You can also use an arrow function like this:
```js
(cb) => {
  request({url, json: true}, cb);
}
```

It is of course also fine to use partials, but I find them often confusing and hard to maintain. In your example you do not know that `request()` expects a `cb` as last parameter.

Comment (soenke-tippelmann) [reply lib/druid.js]: You are right. Funnily I had it like this before, but thought I'd impress the reviewer with my mad javscript skills :D ...

I'll change it back.

**Pull request merged**